clueless/static/js/controllers.js

```diff
@@ -219,19 +240,51 @@ class GameHub {
     this.gamePanel.suggestionQueryPanel.hide();
   }
   handleMsgAccusation(message) {
-    this.gamePanel.showToast('Accusation: ' + JSON.stringify(message));
+    // Get which player has won by character id
+    var player;
+    for (var i = 0; i < gameHub.gameState.players.length; i++) {
+      if (gameHub.gameState.players[i].character.id == message.player) {
+        player = gameHub.gameState.players[i];
+      }
+    }
+    var room = WitnessItem_fromType(message.room, WitnessType.ROOM);
+    var suspect = WitnessItem_fromType(message.suspect, WitnessType.CHARACTER);
+    var weapon  = WitnessItem_fromType(message.weapon, WitnessType.WEAPON);
+
+    var msg = player.display_name + ' made an accusation: ' + room.name + ', ' + suspect.name + ', ' + weapon.name;
+    this.gamePanel.showToast(msg);
   }
   handleMsgWinner(message) {
-    this.gamePanel.showToast('Winner: ' + JSON.stringify(message));
+    // Get which player has won by character id
+    var player;
+    for (var i = 0; i < gameHub.gameState.players.length; i++) {
+      if (gameHub.gameState.players[i].character.id == message.player) {
+        player = gameHub.gameState.players[i];
+      }
+    }
+    var msg = player.display_name + " has won!";
+    this.gamePanel.showToast(msg);
+    this.gamePanel.showModal('Winner!', msg);
   }
   handleMsgDisqualified(message) {
-    this.gamePanel.showToast('Disqualified: ' + JSON.stringify(message));
+    // Get which player was disqualified by character id
+    var player;
+    for (var i = 0; i < gameHub.gameState.players.length; i++) {
+      if (gameHub.gameState.players[i].character.id == message.player) {
+        player = gameHub.gameState.players[i];
+      }
+    }
+    var msg = player.display_name + " has been disqualified.";
+    this.gamePanel.showToast(msg);
+    this.gamePanel.showModal('Disqualified!', msg);
```

Comment:
Does this need to be shown to everyone, since they get the toast msg? Or just if you're the one getting disqualified.

Reply:
I updated so that you only get the modal if you are the player who was disqualified. Also, the message will read "You have been disqualified." Everyone else just gets the toast, not the modal.